src/main/scala/io/iohk/ethereum/blockchain/sync/SyncController.scala

```diff
@@ -47,7 +45,19 @@ class SyncController(
   override def receive: Receive = idle
 
   def idle: Receive = handlePeerUpdates orElse {
-    case StartFastSync =>
+
+    case StartSync =>
+      (appStateStorage.isFastSyncDone(), doFastSync) match {
+        case (false, true) =>
+          self ! StartFastSync
```

Comment:
Should we clear the database before starting fast sync? I got a strange behavior while first setting `doFastSync=false` and then setting it to `true` so that first regular sync is done and then fast sync. This results in:
```
11:15:15.691 i.i.e.blockchain.sync.SyncController - Block: 18816/3359983. Peers: 2/2 (0 blacklisted). State: 17/17 known nodes.
11:15:16.147 i.i.e.b.s.FastSyncReceiptsRequestHandler - Received 64 receipts in 635 ms
11:15:16.677 i.i.e.b.s.FastSyncNodesRequestHandler - Received 256 state nodes in 2452 ms
11:15:17.175 i.i.e.b.s.FastSyncBlockBodiesRequestHandler - Received 128 block bodies in 1014 ms
11:15:17.700 i.i.e.blockchain.sync.SyncController - Block: 192/3359983. Peers: 2/2 (0 blacklisted). State: 273/4369 known nodes.
```
It seems that the blocks obtained from regular sync still remain in storage and will not be used. I don't think this should be an impediment to merging if its too difficult to fix